app/src/main/java/com/portfolio/tasky/usecases/NetworkChangeReceiver.kt

```diff
@@ -0,0 +1,68 @@
+package com.portfolio.tasky.usecases
+
+import android.content.Context
+import android.net.ConnectivityManager
+import android.net.Network
+import android.net.NetworkCapabilities
+import android.net.NetworkRequest
+import androidx.lifecycle.LiveData
+
+
+class NetworkChangeReceiver(var context: Context) : LiveData<NetworkStatus>() {
+
+    val validNetworkConnections : MutableSet<Network> = HashSet()
+    private var connectivityManager: ConnectivityManager =
+        context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
+
+    override fun onActive() {
+        isOnline()
+    }
+
+    override fun onInactive() {
+        connectivityManager.unregisterNetworkCallback(networkCallback)
+    }
+
+    private fun isOnline() {
```

Comment:
Same feedback as in previous PR: This function doesn't tell me what it does. I'd expect that it returns a boolean which it doesn't, so it's full of unknown side-effects from the outside.

Reply:
Removed the function and shifted the code in onActive()

Done